calicoctl/calico_ctl/checksystem.py

```diff
@@ -246,3 +253,37 @@ def _check_etcd_version():
             system_ok = False
 
     return system_ok
+
+
+def _check_kernel_version():
+    """
+    Check that the kernel version meets the minimum required version for Calico.
+    Valid kernel versions have 1 Kernel digit, 1-2 Major digits, and optionally
+    1-2 Minor digits for versioning.  The full version can be followed by at
+    least one non-digit value, then any other characters:
+    Ex.    "2.24.32-100-generic"
+           "3.19.0a-dev"
+           "2.8.1"
+           "3.0"
+           "4.5-rc3"
+    """
+    full_version = platform.uname()[2]
+
+    # Make sure version number makes sense and captures version numbers
+    regex = re.compile("^([\d]\.[\d]{1,2}(\.[\d]{1,2})?)(?:\D.*)?$")
```

Comment:
That could allow you to have version numbers of 2.85789.1 as a valid number.  I think we should limit it to 3 digits max since no number in the linux kernel version has ever been larger than 3 digits (3.4.110 is the largest 3-digit kernel version used).

I'd be happy changing this to:
`^(\d{1,3}(?:\.\d{1,3}){1,3})(?:\D.*)?$`

the `(?:\D.*)?` is important to allow standalone versions like `3.0` or `2.19.26`.

I originally didn't care about capturing the 4th number in an older version string because we don't care about it in this case, but I'll include this for reuse.

What do you think?